### PR TITLE
adding more metrics to the system

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
- [x] Bootstrap time
- [ ] Rpcs latency (per rpc)
- [x] Catchup time
- [x] Transitions in catchup_scheduler

### CODE REVIEW

File: src/lib/perf_histograms/rpc.ml [hunk 1]
```diff
@@ -10,6 +10,10 @@ let decorate_dispatch ~name (dispatch : ('q, 'r) Intf.dispatch) :
   Perf_histograms0.add_span
     ~name:(sprintf "rpc_dispatch_%s" name)
     (Time.diff (Time.now ()) start) ;
+  Coda_metrics.(
+    Gauge.set
+      (Network.rpc_latency_ms ~name)
+      Core.Time.(Span.to_ms @@ diff (now ()) start)) ;
```
Comment (bkase) [inline src/lib/perf_histograms/rpc.ml]: I thought we were saying this happens too frequently to be a Gauge, and we'd need to send the histogram to prometheus -- did I misunderstand that? cc @nholland94

**Comment by bkase** [comment]
Let's land this as is as it's a huge improvement over what we have now, and we can fix the histogram reporting later

File: src/lib/bootstrap_controller/bootstrap_controller.ml [hunk 2]
```diff
@@ -532,6 +532,18 @@ end = struct
             in
             (new_frontier, sorted_filtered_collected_transitins) )
 
+  let run ~logger ~trust_system ~verifier ~network ~frontier ~ledger_db
```
Comment (wu-s-john) [inline src/lib/bootstrap_controller/bootstrap_controller.ml]: I don't like shadowing the `run` function often. Can you not do that here?

File: src/lib/coda_metrics/coda_metrics.ml [hunk 3]
```diff
@@ -234,7 +242,11 @@ module Network = struct
     let help = "# of messages received" in
     Counter.v "messages_received" ~help ~namespace ~subsystem
 
-  (* TODO: track non gossip RPC messages as well *)
+  (*
```
Comment (wu-s-john) [inline src/lib/coda_metrics/coda_metrics.ml]: delete commented code

File: src/lib/coda_metrics/coda_metrics.ml [hunk 4]
```diff
@@ -393,14 +405,19 @@ module Transition_frontier_controller = struct
     in
     Gauge.v "transitions_being_processed" ~help ~namespace ~subsystem
 
-  (* TODO:
   let transitions_in_catchup_scheduler =
-    Counter.v_label "transitions_in_catchup_scheduler" ~namespace ~subsystem
+    let help = "transitions stored inside catchup scheduler" in
+    Gauge.v "transitions_in_catchup_scheduler" ~help ~namespace ~subsystem
 
+  (*
```
Comment (wu-s-john) [inline src/lib/coda_metrics/coda_metrics.ml]: delete commented code

File: src/lib/perf_histograms/rpc.ml [hunk 5]
```diff
@@ -10,6 +10,12 @@ let decorate_dispatch ~name (dispatch : ('q, 'r) Intf.dispatch) :
   Perf_histograms0.add_span
     ~name:(sprintf "rpc_dispatch_%s" name)
     (Time.diff (Time.now ()) start) ;
+  (*
```
Comment (wu-s-john) [inline src/lib/perf_histograms/rpc.ml]: delete comment

File: src/lib/transition_handler/catchup_scheduler.ml [hunk 6]
```diff
@@ -225,7 +234,10 @@ module Make (Inputs : Inputs.S) = struct
           Hashtbl.set t.collected_transitions ~key:parent_hash
             ~data:(cached_transition :: cached_sibling_transitions) ;
           Hashtbl.update t.collected_transitions hash
-            ~f:(Option.value ~default:[])
+            ~f:(Option.value ~default:[]) ;
```
Comment (wu-s-john) [inline src/lib/transition_handler/catchup_scheduler.ml]: Instead of `Hashtbl.update`, should we have Hashtbl.add_exn?

Comment (ghost-not-in-the-shell) [reply src/lib/transition_handler/catchup_scheduler.ml]: No. If that transition has children, we don't want to change its list of children. If it doesn't have any child yet, then we create an entry for it.

**Pull request merged**